File: iosPromotionSDK/PromotionSDKUI/Theme/PromotionThemeStore.swift
//
//  PromotionThemeStore.swift
//  PRMSDK
//
//  Persist theme: serialize PRMSDKTheme → JSON rồi lưu qua `PromotionContainer.preferences`
//  của lõi (cùng cơ chế `PromotionPreferences` mà Android dùng). Tầng này giữ **key** và việc
//  serialize; lõi giữ cơ chế lưu. Đối ứng `PromotionThemeStore.kt`.
//

import Foundation
@_implementationOnly import PRMKotlinBridge

enum PromotionThemeStore {
    private static let keyTheme = "promotion_theme_config_v1"

    static func save(_ theme: PRMSDKTheme) {
        PromotionContainer.shared.preferences.putString(key: keyTheme, value: PRMThemeJson.toJson(theme))
    }

    static func load() -> PRMSDKTheme? {
        guard let json = PromotionContainer.shared.preferences.getString(key: keyTheme) else { return nil }
        return PRMThemeJson.fromJson(json)
    }

    static func clear() {
        PromotionContainer.shared.preferences.remove(key: keyTheme)
    }
}

File: AndroidPromotionSDK/src/main/java/com/ttcn/promotionsdk/promotionsdkui/theme/PromotionThemeStore.kt
package com.ttcn.promotionsdk.promotionsdkui.theme

import com.ttcn.promotionsdk.core.di.PromotionContainer

/**
 * Persist theme: serialize `PRMSDKTheme` → JSON rồi lưu qua `PromotionContainer.putPreference`
 * của lõi (cơ chế chung với iOS). Tầng này giữ **key** và việc serialize; lõi giữ cơ chế lưu.
 * Đối ứng `PromotionThemeStore.swift`.
 */
internal object PromotionThemeStore {
    private const val KEY_THEME = "promotion_theme_config_v1"

    // Fail-open: `preferences` ném nếu chưa initialize() — theme không phải chức năng sống-còn.
    fun save(theme: PRMSDKTheme) {
        runCatching { PromotionContainer.preferences.putString(KEY_THEME, PRMThemeJson.toJson(theme)) }
    }

    fun load(): PRMSDKTheme? =
        runCatching { PromotionContainer.preferences.getString(KEY_THEME) }
            .getOrNull()?.let(PRMThemeJson::fromJson)

    fun clear() {
        runCatching { PromotionContainer.preferences.remove(KEY_THEME) }
    }
}
